### ISSUE TITLE
Volume Reconstruction error when i try to reduce OutputSpacing

### ISSUE THREAD

**Issue by zhaobingshuai** [PlusToolkit/PlusLib#708]
hello everyone, I want to reduce OutputSpacing up to 0.06 mm/pixel when I reconstruction a volume via PLUS.  It is working when OutputSpacing is 1 or bigger. But  It has an error when I try to reduce OutputSpacing.

the following is an error:
```
zhaobingshuai@abing:~/PlusBuild_bin/bin$ ./VolumeReconstructor --config-file=../../us_no_fill_holes.xml --image-to-reference-transform=ImageToProbe --source-seq-file=../../VAS_LINEAR_20200110214433149123456.mhd  --output-volume-file=../../FirstVRResultInLinux34.mha
System start timestamp: 1597103471
Software version: Plus-2.9.0.470eb06d - Linux
Reading configuration file:../../us_no_fill_holes.xml
Reading image sequence ../../VAS_LINEAR_20200110214433149123456.mhd
|WARNING|000.010620| Kinds not found in file: ../../VAS_LINEAR_20200110214433149123456.mhd. Treating the last dimension as time.| in /data/zhaobingshuai/PlusBuild_bin/IGSIO/Source/SequenceIO/vtkIGSIOMetaImageSequenceIO.cxx(199)
Set volume output extent...
ERROR: In /data/zhaobingshuai/PlusBuild_bin/vtk/Common/Core/vtkGenericDataArray.txx, line 404
vtkUnsignedShortArray (0x55748241c7a0): Unable to allocate 396831181691 elements of size 2 bytes. 

Aborted (core dumped)
```

the following is my config file for Volume Reconstruction:

```
<PlusConfiguration version="2.9" PlusRevision="Plus-2.0.0.2801">
  <VolumeReconstruction
    OutputSpacing="0.06 0.06 0.06"
    Interpolation="LINEAR"
    Optimization="FULL"
    CompoundingMode ="MEAN"
    FillHoles="OFF"
	NumberOfThreads="16">
  </VolumeReconstruction>
</PlusConfiguration>
```

this a link of my input file:
https://www.dropbox.com/sh/c1qmeg7trmfu11t/AAAbWlhIPuDVp2gVqN-ghQ9la?dl=0
input file's size is 605*672*378=153679680 voxels, According to my calculations, the occupied memory should be 293 MB instead of 739 GB.
At first, I thought that the reason might be insufficient memory on my Windows laptop, so I compiled PLUS under Linux, and my Linux workstation has 4 GeForce RTX2080Ti GPUs. But the error still persists.

Hope someone can tell me why? And  What should I do to get the correct output at a smaller OutputSpacing(eg. 0.05)?  Thanks for your attention.

**Comment by adamrankin** [comment]
This doesn't use GPU memory, it uses system memory.

The output size is calculated using the input images and their positions. The final volume size, at a spacing of 0.05, seems to be exceeding your available memory.

Some proposed solutions from http://perk-software.cs.queensu.ca/plus/doc/nightly/user/AlgorithmVolumeReconstruction.html
```
* Perform a low-resolution "scout scan" reconstruction from file (e.g., with 3mm spacing), define the region of interest, 
and then do high-resolution live reconstruction. During live reconstruction the input frames do not have to be kept in 
memory, which allows reconstruction of higher-resolution volumes. This workflow is implemented in PlusRemote with 
a nice graphical user interface (available as a 3D Slicer module in the SlicerOpenIGTLink extension)
* Reduce the region of interest size (change OutputExtent, OutputSpacing, and OutputOrigin)
```

**Comment by adamrankin** [comment]
You can also increase your pagefile to ~1TB at the cost of performance.

**Comment by zhaobingshuai** [comment]
I want the lesions in the output of volume reconstruction to have the same length as the real lesions, so I had to set OutputSpacing to 0.06. I found that: for an input of [10,20,30], the output Dimension of volume reconstruction is the input Dimension /OutputSpacing([10/OutputSpacing,20/OutputSpacing,30/OutputSpacing]=[166,333,500]), the corresponding size increases from 10*20*30=6000 of the input to 2763900 of the output, an increase of 460 times. I guess this is the cause of insufficient memory. In fact, I only need OutputSpacing to be 0.06, and the output dimension should remain the same.
What should I do to change OutputSpacing without changing the output dimension? Or how is this specific...


> You can also increase your pagefiles/swap space to ~1TB at the cost of performance.

Thank you.
